docs/data/material/components/stacked-snackbars/MaximumSnackbars.tsx

```diff
@@ -0,0 +1,55 @@
+import * as React from 'react';
+import SnackbarsProvider from '@mui/lab/SnackbarsProvider';
+import useSnackbars from '@mui/lab/useSnackbars';
+import Button from '@mui/material/Button';
+import ButtonGroup from '@mui/material/ButtonGroup';
+import Stack from '@mui/material/Stack';
+
+function MyApp({
+  snackbarLimit,
+  setSnackbarLimit,
+}: {
+  snackbarLimit: number;
+  setSnackbarLimit: (fn: (prevState: number) => number) => void;
+}) {
+  const snackbars = useSnackbars();
+
+  return (
+    <Stack spacing={2}>
+      <ButtonGroup sx={{ justifyContent: 'center' }}>
+        <Button
+          disabled={snackbarLimit === 1}
+          onClick={() => setSnackbarLimit((prevState) => prevState - 1)}
+        >
+          -
+        </Button>
+        <Button disableRipple>{snackbarLimit}</Button>
+        <Button
+          disabled={snackbarLimit === 10}
+          onClick={() => setSnackbarLimit((prevState) => prevState + 1)}
+        >
+          +
+        </Button>
+      </ButtonGroup>
+      <Button
+        onClick={() =>
+          snackbars.showSnackbar({
+            message: 'Note archived',
+            autoHideDuration: 4000,
+          })
+        }
+      >
+        Show Snackbar
+      </Button>
+    </Stack>
+  );
+}
+
+export default function MaximumSnackbars() {
+  const [snackbarLimit, setSnackbarLimit] = React.useState<number>(2);
+  return (
+    <SnackbarsProvider limit={snackbarLimit}>
+      <MyApp snackbarLimit={snackbarLimit} setSnackbarLimit={setSnackbarLimit} />
```

Comment:
I think that we can improve this demo. We can remove the -/+ buttons

<img width="155" alt="Screenshot 2022-05-23 at 01 29 39" src="https://user-images.githubusercontent.com/3165635/169720687-fe9c7316-8312-400d-9650-5b22eed893ed.png">
 
It would make it much easier to find the interesting part of this demo. Ideally, we would have this:

Then, for developers who want to try different values: #32107 solves the problem. I would argue the DX would be much better. See

![May-23-2022 01-36-38](https://user-images.githubusercontent.com/3165635/169720912-3921eae6-0176-46ff-babd-b8ff8d0e8f34.gif)

Reply:
Do we need to optimize for non-devs? I mean, would a designer want to play around with this demo? I think that we should have the backend devs as our primary target when we write the documentation. For them, what would yield the best DX?

> I think I'd wait for that PR to be merged in order to remove the interactive buttons here.

Opening a new GitHub issue to do clean up sounds great; I think that there will be a good number of interactive demos in our docs that we can remove (improve) thanks to https://github.com/mui/material-ui/pull/32107, this is exciting! :)